Topics/Tech_Stacks/ORM.md

```diff
@@ -0,0 +1,127 @@
+# ORM (Object Relation Mapper)
+
+## Table of Content
+### [Introductions](#introductions-1)
+### [What is an ORM](#what-is-an-orm-1)
+### [Example](#example-1)
+### [Should I use one?](#should-i-use-one-1)
+### [Advantages](#advantages-1)
+### [Disadvantages](#disadvantages-1)
+### [Conclusion](#conclusion-1)
+
+## Introductions:
+Most applications that you will build, in fact most applications that exist today, require some sort of data permanence. Meaning that data is stored somewhere, and is not lost when a site/app is reloaded. We employ databases like MYSQL, PostgreSQL, and MongoDB to ensure that our data can be stored and we can safely interact with it. But how do we do that? There are many database drivers that can allow us to write queries as lines of code directly from within our javascript or python (or whatever language you prefer) scripts. On the other hand, you may consider using an ORM to write your database code!
+
+
+## What is an ORM:
+
+An ORM (Object Relation Mapper) is a software tool that allows for the translation of an entity between its representation in a database (A relation) and object-oriented programming languages (An Object). As such, they allow us to simply treat all data as objects in our code, and leverage the ORM when we want to store this object in our relational database of choice. 
+
+- Its good to node that ORM alternatives exist for non-relational databases too, these would be ODMs (Object Document Models), for example, MongoDB has one called mongoose which applies to javascript/typescript. A lot of these discussions apply to these ODMs too. 
+
+## Example:
+Let's take a simple example, where we want to connect to a database, create a user table, and add a new user to our database. How would that look as SQL Queries compared to an ORM? 
```

Comment:
I think it would be helpful to state what language the example is in, and also that ORMs exist for many different languages.